### ISSUE TITLE
Syntax errors in .d.ts file (unassigned const export, overloaded function type)

### ISSUE THREAD

**Issue by bpstrngr** [TyrealHu/acorn-typescript#37]
rollup/src/rollup/types.d.ts: 
```
export const VERSION: string;
                            ^
```
```
SyntaxError: Unexpected token (1:28)
    at pp.raise (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/haverbeke_2012_acorn/0/acorn/src/location.js:15:13)
    at TypeScriptParser.raiseCommonCheck (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/tyrealhu_2023_acorn_typescript.js:4975:44)
    at TypeScriptParser.raise (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/tyrealhu_2023_acorn_typescript.js:4981:29)
    at pp.unexpected (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/haverbeke_2012_acorn/0/acorn/src/parseutil.js:111:8)
    at pp.parseVar (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/haverbeke_2012_acorn/0/acorn/src/statement.js:507:12)
    at TypeScriptParser.parseVarStatement (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/tyrealhu_2023_acorn_typescript.js:3519:23)
    at pp.parseStatement (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/haverbeke_2012_acorn/0/acorn/src/statement.js:116:17)
    at TypeScriptParser.parseStatement (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/tyrealhu_2023_acorn_typescript.js:3568:30)
    at TypeScriptParser.parseExport (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/xtuc_2020_acorn_importassertions/0/src/index.js:101:33)
    at TypeScriptParser.parseExport (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/tyrealhu_2023_acorn_typescript.js:3392:34) {
  pos: 28,
  loc: Position { line: 1, column: 28 },
  raisedAt: 29
}
```
last in a batch of 4 issues.

**Comment by TyrealHu** [comment]
Its not an issue to `acorn-typescript`. U have to enable dts while parsing `.d.ts` file
for example
```
acorn.Parser.extend(tsPlugin({
  dts: true
}))
```

**Comment by bpstrngr** [comment]
right, makes sense. 
I added the condition to my loader: 
```
  if(dialect=="typescript")
  Parser=await import("./tyrealhu_2023_acorn_typescript.js").then(({default:plugin})=>
  Parser.extend(plugin({dts:options.source.endsWith(".d.ts")})));
```
Now i get error only for the last line of that file: 
```
export function defineConfig(options: RollupOptions): RollupOptions;
export function defineConfig(options: RollupOptions[]): RollupOptions[];
```
```
SyntaxError: Duplicate export 'defineConfig' (929:16)
    at pp.raise (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/haverbeke_2012_acorn/0/acorn/src/location.js:15:13)
    at TypeScriptParser.raiseCommonCheck (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/tyrealhu_2023_acorn_typescript.js:4990:44)
    at TypeScriptParser.raiseRecoverable (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/tyrealhu_2023_acorn_typescript.js:4993:29)
    at pp.checkExport (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/haverbeke_2012_acorn/0/acorn/src/statement.js:930:10)
    at TypeScriptParser.parseExport (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/xtuc_2020_acorn_importassertions/0/src/index.js:105:16)
    at TypeScriptParser.parseExport (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/tyrealhu_2023_acorn_typescript.js:3392:34)
    at pp.parseStatement (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/haverbeke_2012_acorn/0/acorn/src/statement.js:137:69)
    at TypeScriptParser.parseStatement (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/tyrealhu_2023_acorn_typescript.js:3568:30)
    at pp.parseTopLevel (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/haverbeke_2012_acorn/0/acorn/src/statement.js:22:21)
    at TypeScriptParser.parse (file:///mnt/chromeos/GoogleDrive/MyDrive/blik/haverbeke_2012_acorn/0/acorn/src/state.js:97:17) {
  pos: 26216,
  loc: Position { line: 929, column: 16 },
  raisedAt: 26273
}
```
I'll repurpose this issue to address this dts type overloading.

**Comment by TyrealHu** [comment]
fixed in version 1.4.4
